corgi/tasks/prod_defs.py

```diff
@@ -63,192 +63,203 @@ def update_products() -> None:
             )
 
             for pd_product_version in pd_product_versions:
-                pd_product_streams = pd_product_version.pop("product_streams", [])
+                parse_product_version(pd_product_version, product, product_node)
+            product.save_product_taxonomy()
 
-                name = pd_product_version.pop("id")
-                if match_version := RE_VERSION_LIKE_STRING.search(name):
-                    version = match_version.group()
-                description = pd_product_version.pop("public_description", [])
 
-                logger.debug(
-                    "Creating or updating Product Version: name=%s, description=%s",
-                    name,
-                    description,
-                )
-                product_version, _ = ProductVersion.objects.update_or_create(
-                    name=name,
-                    defaults={
-                        "version": version,
-                        "description": description,
-                        "products": product,
-                        "meta_attr": pd_product_version,
-                    },
-                )
-                product_version_node, _ = ProductNode.objects.get_or_create(
-                    object_id=product_version.pk,
-                    defaults={
-                        "parent": product_node,
-                        "obj": product_version,
-                    },
-                )
+def parse_product_version(pd_product_version, product, product_node):
+    pd_product_streams = pd_product_version.pop("product_streams", [])
+    name = pd_product_version.pop("id")
```

Comment:
We already use the variable name product_version below for the new object. Also we already use the name pd_product_version as the dict object parse from the JSON, so we are running out of options for names. However the I think the current name is acceptable because the focus of the function is on parsing a product version, so that should be enough context to figure out what we are naming.